GoogleForCreators/web-stories-wp
# Dashboard: Enable arrow navigation on explore templates grid

## Summary
Allows users to key into the templates grid and use arrow keys to navigate. 

## Relevant Technical Choices

- Follows same pattern as here: https://github.com/google/web-stories-wp/pull/4303 
- Adds an invisible button to each grid item that will transfer focus to grid items according to `useGridViewKeys` without interrupting the grid structure and preventing nested buttons. 
- when new `FocusableGridItem` is set a useEffect hook immediately updates the focus to the actual grid item.

## To-do
- implement on my stories
- return focus to proper item after closing the preview dialog
- Sam's UI updates to the preview button 
- It'll be nice to add more karma tests to this when we're not on a deadline.


## User-facing changes

Now when you are on explore templates and tab to the grid you can enter the grid items by using arrow keys or tab again to exit the grid and, if you are not at the top of the browser window, jump into the scroll to top button.
New text for template actions: 
- "View" => "See Details" 
- invisible preview where you would click anywhere not a button on the item => "Preview" (this will change pretty fast, Sam's working on that)

## Testing Instructions

- Navigate to 'explore templates' in dashboard
- Use 'Tab' to change focused element until you get to the grid. The grid should have a blue outline now. 
  - Hit tab again to exit the entire grid (this can be tricky to see if the scroll to top button is not in view, but it should go there). 
  - Hit the right arrow key and enter into the grid. Now you're focused on that first template and if you tab again you'll see the 'preview' button gain focus, tab again and get the 'see details' button in focus. One more tab and you get the 'use template' button. Tab again,  you exit the grid (get sent to the scroll to top button) OR instead of hitting tab that fourth time hit another arrow key and see focus shift to that new grid item. 
- If you exit the grid and hit shift + tab, focus will be regained to whatever grid item you left off on. 
- This same functionality should persist for the detail template view related grids section. 
---



Addresses #4296
